### DISCUSSION TITLE
Example for an e-commerce product view with jetpack compose

### DISCUSSION THREAD

**Discussion by kishaningithub** [sceneview/sceneview#268] [Q&A]
It would be great to have an example for creating an e-commerce AR product view component using jetpack compose

**Comment by JohanAlbrectsen** [comment]
Yes, I agree

**Comment by Blizl** [comment]
I have a PR up for this https://github.com/SceneView/sceneview-android/pull/300

**Comment by ThomasGorisse** [comment]
Here's a product viewer in v4.0.0-rc.1:

```kotlin
implementation("io.github.sceneview:sceneview:4.0.0-rc.1")

@Composable
fun ProductViewer(modelUrl: String) {
    SceneView(
        cameraManipulator = rememberCameraManipulator(
            orbitHomePosition = Position(0f, 0f, 4f)
        )
    ) {
        val model = rememberModelInstance(modelLoader, modelUrl)
        if (model != null) {
            ModelNode(
                modelInstance = model,
                scaleToUnits = 1.0f
            )
        }
    }
}
```

This gives you orbit camera, pinch-to-zoom, and model auto-scaling out of the box. See `samples/android-demo` for a full Material 3 showcase with model switching.

**Comment by ThomasGorisse** [comment]
Here's a Compose e-commerce product viewer with SceneView v4:

```kotlin
@Composable
fun ProductViewer(modelUrl: String) {
    val engine = rememberEngine()
    val modelLoader = rememberModelLoader(engine)
    val model = rememberModelInstance(modelLoader, modelUrl)

    SceneView(
        modifier = Modifier.fillMaxSize(),
        engine = engine,
        modelLoader = modelLoader
    ) {
        if (model != null) {
            ModelNode(
                modelInstance = model,
                scaleToUnits = 1f,
                centerOrigin = Position(y = -0.5f)
            )
        }
    }
}
```

The camera gesture system (orbit, zoom, pan) works out of the box — no extra config needed. See the [3D Model Viewer recipe](https://github.com/sceneview/sceneview/blob/main/samples/recipes/model-viewer.md) for a complete example with lighting, environment, and animations.
